client/asset/dcr/dcr.go

```diff
@@ -987,9 +999,12 @@ func (dcr *ExchangeWallet) FindRedemption(ctx context.Context, coinID dex.Bytes)
 
 // Refund refunds a contract. This can only be used after the time lock has
 // expired.
-func (dcr *ExchangeWallet) Refund(receipt asset.Receipt, nfo *dex.Asset) error {
-	op := receipt.Coin()
-	txHash, vout, err := decodeCoinID(op.ID())
+// NOTE: The contract cannot be retreived from the unspent coin info as the
+// wallet does not store it, even though it was known when the init transaction
+// was created. The DEX should store this information for persistence across
+// sessions.
+func (dcr *ExchangeWallet) Refund(coinID, contract dex.Bytes, nfo *dex.Asset) error {
```

Comment:
This is a good move. Since the exchange wallets are arbitrating refundability via `LockTimeExpired` now, we can probably drop the `asset.Receipt` interface altogether and just use `asset.Coin`s.